### PR TITLE
fix time desync

### PR DESCRIPTION
**Pull request by sgaragagghu**

### CODE REVIEW

**Comment by jonano614** [comment]
What does this fix fixes?

**Comment by sgaragagghu** [comment]
> What does this fix fixes?
the desync when there is a pool with bad clock

**Comment by sgaragagghu** [comment]
> What does this fix fixes?

the desync when there is a pool with bad clock

File: client/init.h [hunk 1]
```diff
@@ -17,10 +17,14 @@ enum xdag_states
...
-    xdag_diff_t difficulty, max_difficulty;
-    uint64_t nblocks, total_nblocks;
-    uint64_t nmain, total_nmain;
-    uint32_t nhosts, total_nhosts, reserved1, reserved2;
+	xdag_diff_t difficulty, max_difficulty;
+	uint64_t nblocks, total_nblocks;
+	uint64_t nmain, total_nmain;
+	uint32_t nhosts, total_nhosts;
+	union {
+		uint32_t reserved1, reserved2;
```
Comment (xrdavies) [inline client/init.h]: Should not be 
```
union {
	struct {
		uint32_t reserved1, reserved2;
	};
 	uint64_t main_time;
 };
```
?

File: client/transport.c [hunk 2]
```diff
@@ -67,19 +70,21 @@ static int process_transport_block(struct xdag_block *received_block, void *conn
 	xtime_t start_time = xdag_start_main_time();
 	xtime_t current_time = xdag_main_time();
 
-	if(current_time < start_time || stats->total_nmain > current_time - start_time + 2) return -1;
+	if(current_time >= start_time && stats->total_nmain <= current_time - start_time + 2) {
+		if(stats->main_time <= current_time + 2) {
```
Comment (xrdavies) [inline client/transport.c]: Here we suppose the latency of the network is 2 seconds?

File: client/init.h (hunk 1 above)
Comment (sgaragagghu) [reply client/init.h]: right, fixed

File: client/transport.c (hunk 2 above)
Comment (sgaragagghu) [reply client/transport.c]: yes, i left it because it was already 2 seconds, we can test and reduce it

**Comment by jonano614** [comment]
No more comments from my side too. Thank you for your efforts!

**Pull request merged**